src/ensembl/src/content/app/entity-viewer/shared/components/entity-viewer-sidebar/entity-viewer-sidebar-modal/modal-views/EntityViewerSidebarBookmarks.tsx

```diff
@@ -0,0 +1,171 @@
+  const exampleGene = props.exampleEntities.find(({ type }) => type === 'gene');
+
+  if (!exampleGene?.id) return <></>;
+
+  const featureIdInUrl = buildFocusIdForUrl({
+    type: 'gene',
+    objectId: exampleGene.id
+  });
+  const path = urlFor.entityViewer({
+    genomeId: props.activeGenomeId,
+    entityId: featureIdInUrl,
+    view: 'transcripts'
+  });
+
+  return (
+    <div data-test-id="example links">
+      <div className={styles.sectionTitle}>Example links</div>
+      <div key={exampleGene.id} className={styles.linkHolder}>
+        <Link to={path} onClick={closeSidebarModal}>
+          Example {exampleGene.type}
+        </Link>
+      </div>
+    </div>
+  );
+};
+
+type PreviouslyViewedLinksProps = {
+  genomeId: string;
+  entityId: string;
+  previouslyViewedEntities: ReturnType<typeof getPreviouslyViewedEntities>;
+  closeSidebarModal: () => void;
+};
+
+export const PreviouslyViewedLinks = (props: PreviouslyViewedLinksProps) => {
+  const onLinkClick = (objectType: string, index: number) => {
+    analyticsTracking.trackEvent({
+      category: 'recent_bookmark_link',
+      label: objectType,
+      action: 'clicked',
+      value: index + 1
+    });
+
+    props.closeSidebarModal();
+  };
+
+  const activeObjectId = parseEnsObjectId(props.entityId);
+  const previouslyViewedEntitiesWithoutActiveObject = props.previouslyViewedEntities.filter(
+    (entity) => entity.stable_id !== activeObjectId.objectId
+  );
+
+  return (
+    <div data-test-id="previously viewed links">
+      {[...previouslyViewedEntitiesWithoutActiveObject]
+        // .reverse()
+        .map((previouslyViewedEntity, index) => {
+          const path = urlFor.entityViewer({
+            genomeId: props.genomeId,
+            entityId: buildFocusIdForUrl({
+              type: 'gene',
+              objectId: previouslyViewedEntity.stable_id
+            })
+          });
+
+          return (
+            <div
+              key={previouslyViewedEntity.label}
+              className={styles.linkHolder}
+            >
+              <Link
+                to={path}
+                onClick={() => onLinkClick(previouslyViewedEntity.label, index)}
+              >
+                {previouslyViewedEntity.label}
+              </Link>
+              <span className={styles.previouslyViewedType}>
+                {upperFirst(previouslyViewedEntity.type)}
+              </span>
+            </div>
+          );
+        })}
+    </div>
+  );
+};
+
+export const EntityViewerSidebarBookmarks = () => {
+  const activeGenomeId = useSelector(getEntityViewerActiveGenomeId) || '';
+  const activeEntityId = useSelector(getEntityViewerActiveEnsObjectId) || '';
+
+  const exampleEntities = useSelector((state: RootState) =>
+    getGenomeExampleFocusObjects(state, activeGenomeId)
+  );
+  const previouslyViewedEntities = useSelector((state: RootState) =>
+    getPreviouslyViewedEntities(state, activeGenomeId)
+  );
+
+  const dispatch = useDispatch();
+
```

Comment:
probably at this point we should return `null` from the component if `activeGenomeId === ''` or `activeEntityId === ''`. Down below you are passing activeEntityId to the PreviouslyViewedLinks component, by which point `activeEntityId` better not be empty string.